src/contracts/Erc20Contract.ts

```diff
@@ -1,29 +1,29 @@
-import BN from 'bn.js'
-
-import { TransactionObject } from 'contracts/types'
-import { Contract } from 'web3-eth-contract'
-
-export interface Erc20Contract extends Contract {
-  clone(): Erc20Contract
-
-  methods: {
-    totalSupply(): TransactionObject<string>
-    decimals(): TransactionObject<string>
-    symbol(): TransactionObject<string>
-    name(): TransactionObject<string>
-
-    balanceOf(owner: string): TransactionObject<string, [string]>
-
-    allowance(owner: string, spender: string): TransactionObject<string, [string, string]>
-
-    approve(spender: string, value: number | string | BN): TransactionObject<boolean, [string, string | number | BN]>
+import { EventOptions } from 'web3-eth-contract'
+import { EventLog } from 'web3-core'
+
+// Doesn't use ContractEvent on purpose since the mapping of events in typechain is broken
+// TODO: Update when typechain fix the issue
+import { /* ContractEvent, */ Callback } from './gen/types'
+import { Erc20 } from './gen/Erc20'
+import { ContractEvent } from './types'
+
+export interface Approval {
+  owner: string
+  spender: string
+  value: string
+}
 
-    transfer(to: string, value: number | string | BN): TransactionObject<boolean, [string, string | number | BN]>
+export interface Transfer {
+  from: string
+  to: string
+  value: string
+}
 
-    transferFrom(
-      from: string,
-      to: string,
-      value: number | string | BN,
-    ): TransactionObject<boolean, [string, string, string | number | BN]>
+export interface Erc20Contract extends Omit<Erc20, 'events'> {
+  // Redefine "events" because it's not correctly generated using typechain
```

Comment:
I don't immediately see what is wrong with typechain events. Is it only `allEvents`?

Reply:
Notice how I import `ContractEvent` from `types` and not `'./gen/types'`.

They define `EventOptions` and I get it from `web3-eth-contract`

But I might have something else wrong, cause I'm getting an error when I try to use this package, and it's for the events.

Reply:
That's unfortunate. Can you create an issue in their repo

Reply:
I don't have capacity right now, I've created this to create them a PR or to open an issue https://github.com/gnosis/dex-js/issues/31